services/tenant-ui/frontend/src/components/verifier/Presentations.vue

```diff
@@ -3,32 +3,40 @@
 
   <ProgressSpinner v-if="loading" />
   <div v-else>
-    <DataTable v-model:selection="selectedPresentation" :value="presentations" :paginator="true" :rows="10" striped-rows selection-mode="single">
+    <DataTable v-model:selection="selectedPresentation" v-model:expandedRows="expandedRows" 
+       :value="presentations" dataKey="verifier_presentation_id" :paginator="true" 
+       :rows="10" striped-rows selection-mode="single"  @rowExpand="onRowExpand">
+      <Column :expander="true" headerStyle="width: 3rem" />
       <Column :sortable="true" field="name" header="Name" />
       <Column field="contact.alias" header="Contact Name" />
       <Column field="state" header="State" />
       <Column field="status" header="Status" />
       <Column field="created_at" header="Created at" />
+      <template #expansion="{data, index}">
+          <PresentationDetails :presentation="presentationDetailDict[expandedRows[index].verifier_presentation_id]"/>
+      </template>
     </DataTable>
   </div>
 </template>
 
 
 <script setup lang="ts">
-import { onMounted } from 'vue';
+import { onMounted, ref } from 'vue';
 import Column from 'primevue/column';
-import DataTable from 'primevue/datatable';
+import DataTable, { DataTableRowCollapseEvent, DataTableRowExpandEvent } from 'primevue/datatable';
 import ProgressSpinner from 'primevue/progressspinner';
 import { useToast } from 'vue-toastification';
 
 import { useVerifierStore } from '../../store';
 import { storeToRefs } from 'pinia';
 
+
+import PresentationDetails from './PresentationDetails.vue';
 const toast = useToast();
 
 const verifierStore = useVerifierStore();
 // use the loading state from the store to disable the button...
-const { loading, presentations, selectedPresentation } = storeToRefs(useVerifierStore());
+const { loading, presentations, selectedPresentation, presentationDetailDict} = storeToRefs(useVerifierStore());
```

Comment:
I think this is a very good thing to keep in the store, these are expensive api calls if they need to referred to later, they are stored in an easy way. If you need to refresh that specific object. just call the GET again with the ID and it will override the existing one. 

I'm confident this pattern will scale work for single items, or many items, while being readable. It certainly could be managed within the component, but we have a store, we should use it. Having free floating objects scattered within components sounds more error prone then establishing that loaded items should always go in the store, and it's easy to mimic.

Reply:
chances are slim that there will be 2 simultaneous "binds" to items in that dict, but... having that in the store is way more prone to errors. if I just ask to see something from the store by id (because it already exists), the data could be very stale. if I refresh it with different parameters, the data is now different than another component expects.  if the only way to know it is fresh and what you want, you need to fetch it, so cost savings of the call are gone. i see very limited upside and a whole lot of down.

Reply:
I see a point about the parameters, but that can be checked at a component level if it's important. If you concern is around stale data, then that is true for everything that is loaded into the store. I don't see how this usage is any more prone to that than other datasets. 

As you point out, if no other component ends up using this location in the store, then there is no issue. If another component uses the same location, they can easily inspect what component is using it and decide if it aligns enough for reuse, or just refresh, or if the usage is too different, then that new component will make a new location in the store (or store it locally). 

I see very low risk of downside, and potential savings later, could be 0 benefit, but I think it's worth trying.

Reply:
so you think each component should go check all other components that may use that and see if they have conflicts in parameters? i don't think that's a good idea. and again, if the only way to know the data is good is to fetch it then storing it is bad.  if only 1 component uses that, the data still remains until the app is refreshed no? not when the component goes away. so the weight of the app increases which is a downside.

Reply:
if your concern is how expensive the API call is (and very right to be concerned)... shouldn't that be addressed at the Traction level? like a 304 Not Modified?

Reply:
And I believe in a previous PR there were notes and comments about determining which collections should stay in the store and which ones we don't want. So the goal is to reduce those when there is no benefit. We will probably only keep things that will appear in dropdown lists.

Reply:
I'm saying when if and when another component needs this data, i'm betting it will be beneficial that the pieces are already in place, and if it's not, then there is no penalty. 

I am not concerned about the memory footprint of our application, we aren't loading a million entries onto the users machine, when the user clicks on a different tab and comes back, the data will already be loaded, which is a benefit. A force-refresh button is a good idea, and I can add that if needed.

/verifier/presentations/<id> only has the 'acapy' parameters, so there are no issues around different options being set.

Reply:
well, it should have timeline, so file that as a bug. and it does have deleted. 
here's a compromise... change that fetchItem method to not include the dict/map, so that it can be used elsewhere. create a NEW store specifically for your use case, you can use the fetch to populate it and read as needed. you can then create very specific actions and getters that will keep that collection tight and contain exactly what you need and anyone using that knows exactly how to get items and refresh those items. if we need that pattern for other screens/components, it's easy to adopt.

Reply:
may want to call it something like presentationRequestCache? something so it's clear what the use case is.

Reply:
and put in get by id that will check the cache and fetch if needed, also if record is in cache, you could check the other store for timestamp and see if it needs refreshing. that would be a super useful pattern.

Reply:
How about those most recent changes, making it still centralised and useful, but indicated it's less 'generic' as a global solution than 'fetchitem'.

Reply:
i gotta download the code, can't follow through the github UI...

Reply:
hmm...
like there should be a fetchItem call without the storage built in to the call. that can be used repeatedly everywhere, including to build a cache. and I don't think the cache/dict object needs to be exposed or ref'd externally.  you can certainly keep the cache in that store, but use getters/actions to get data out of it.  the fact that it's cached wouldn't be bound outside of the verifierStore.  so the component (Presentations.vue in this case) would get the id (as you have) and simply call verifierStore.getPresentationDetailsFromCache or whatever. internally, that getter would check to see if needs to get it from the server or not (check if it has cached the id, check if the cache is the most recent compared to the collection, fetch if neither is true). the store can have the two methods: getPresentationDetails and getPresentationDetailsFromCache. The first always calls the server and returns the item (with whatever params - timeline, deleted, acapy), the second only calls the server when needed and stores the same item shape in the cache always.